### PR TITLE
Add email support

### PR DESCRIPTION
**Pull request by MatMaul**

### CODE REVIEW

**Comment by MatMaul** [comment]
Should I make sending logs&screenshot as attachments an option ?

File: submit.go [hunk 1]
```diff
@@ -570,6 +586,41 @@ func buildGithubIssueRequest(p parsedPayload, listingURL string) github.IssueReq
 	}
 }
 
+func (s *submitServer) sendEmail(p parsedPayload, reportDir string) error {
+	if s.cfg.SMTPServer == "" {
+		log.Println("Email submission disabled")
```
Comment (richvdh) [inline submit.go:591]: feels like this is a bit redundant and will be a bit noisy on most servers?

File: submit.go [hunk 2]
```diff
@@ -570,6 +586,41 @@ func buildGithubIssueRequest(p parsedPayload, listingURL string) github.IssueReq
...
+	} else {
```
Comment (richvdh) [inline submit.go:592]: an early return here would be nice, to save the big indented block.

**Review by richvdh — APPROVED** [review]
lgtm!

File: submit.go (hunk 1 above)
Comment (MatMaul) [reply submit.go:591]: agreed, but I went for consistency because it does that for GH issues notifs or Slack one. Should I remove them on the way ?

Comment (richvdh) [reply submit.go:591]: yeah, probably.

File: submit.go [hunk 3]
```diff
@@ -470,70 +471,69 @@ func (s *submitServer) saveReport(ctx context.Context, p parsedPayload, reportDi
...
 
+	if err := s.sendEmail(p, reportDir); err != nil {
+		return nil, err
+	}
+
 	return &resp, nil
 }
 
 func (s *submitServer) submitGithubIssue(ctx context.Context, p parsedPayload, listingURL string, resp *submitResponse) error {
-	if s.ghClient == nil {
```
Comment (richvdh) [inline submit.go:477]: I'm failing to see where we now handle the cases of github/slack/email being disabled? we seem to call `submitGithubIssue` unconditionally so won't it just fail?

**Review by richvdh — CHANGES_REQUESTED** [review]
as above

File: submit.go (hunk 3 above)
Comment (MatMaul) [reply submit.go:477]: ... Indeed. Fixing that.

File: rageshake.sample.yaml [hunk 4]
```diff
@@ -20,3 +20,17 @@ github_project_mappings:
 # a Slack personal webhook URL (https://api.slack.com/incoming-webhooks), which
 # will be used to post a notification on Slack for each report.
 slack_webhook_url: https://hooks.slack.com/services/TTTTTTT/XXXXXXXXXX/YYYYYYYYYYY
+
+# notification can also be pushed by email.
+# this param controls the target emails
+email_addresses:
+- support@matrix.org
```
Comment (richvdh) [inline rageshake.sample.yaml:27]: I think this is clearer:

```suggestion
  - support@matrix.org
```

**Review by richvdh — APPROVED** [review]
lgtm otherwise

**Pull request merged**